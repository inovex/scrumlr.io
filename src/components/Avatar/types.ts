export const AVATAR_SKIN_COLORS = ["Tanned", "Yellow", "Pale", "Light", "Brown", "DarkBrown", "Black"] as const;
export type AvatarSkinColor = (typeof AVATAR_SKIN_COLORS)[number];

export const AVATAR_TOP_TYPES = [
  "Eyepatch",
  "Hat",
  "Hijab",
  "LongHairBigHair",
  "LongHairBob",
  "LongHairBun",
  "LongHairCurly",
  "LongHairCurvy",
  "LongHairDreads",
  "LongHairFrida",
  "LongHairFro",
  "LongHairFroBand",
  "LongHairMiaWallace",
  "LongHairNotTooLong",
  "LongHairShavedSides",
  "LongHairStraight",
  "LongHairStraight2",
  "LongHairStraightStrand",
  "NoHair",
  "ShortHairDreads01",
  "ShortHairDreads02",
  "ShortHairFrizzle",
  "ShortHairShaggyMullet",
  "ShortHairShortCurly",
  "ShortHairShortFlat",
  "ShortHairShortRound",
  "ShortHairShortWaved",
  "ShortHairSides",
  "ShortHairTheCaesar",
  "ShortHairTheCaesarSidePart",
  "Turban",
  "WinterHat1",
  "WinterHat2",
  "WinterHat3",
  "WinterHat4",
] as const;
export type AvatarTopType = (typeof AVATAR_TOP_TYPES)[number];

export const AVATAR_CLOTHE_COLORS = [
  "Black",
  "Blue01",
  "Blue02",
  "Blue03",
  "Gray01",
  "Gray02",
  "Heather",
  "PastelBlue",
  "PastelGreen",
  "PastelOrange",
  "PastelRed",
  "PastelYellow",
  "Pink",
  "Red",
  "White",
] as const;
export type AvatarClotheColor = (typeof AVATAR_CLOTHE_COLORS)[number];

export const AVATAR_GRAPHIC_TYPES = ["Bat", "Cumbia", "Deer", "Diamond", "Hola", "Pizza", "Resist", "Selena", "Bear", "SkullOutline", "Skull"] as const;
export type AvatarGraphicType = (typeof AVATAR_GRAPHIC_TYPES)[number];

export const AVATAR_CLOTHE_TYPES = ["BlazerShirt", "BlazerSweater", "CollarSweater", "GraphicShirt", "Hoodie", "Overall", "ShirtCrewNeck", "ShirtScoopNeck", "ShirtVNeck"] as const;
export type AvatarClotheType = (typeof AVATAR_CLOTHE_TYPES)[number];

export const AVATAR_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "PastelPink", "Blue", "Platinum", "Red", "SilverGray"] as const;
export type AvatarHairColor = (typeof AVATAR_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "Platinum", "Red"] as const;
export type AvatarFacialHairColor = (typeof AVATAR_FACIAL_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_TYPES = ["Blank", "BeardMedium", "BeardLight", "BeardMajestic", "MoustacheFancy", "MoustacheMagnum"] as const;
export type AvatarFacialHairType = (typeof AVATAR_FACIAL_HAIR_TYPES)[number];

export const AVATAR_ACCESSORIES_TYPES = ["Blank", "Kurt", "Prescription01", "Prescription02", "Round", "Sunglasses", "Wayfarers"] as const;
export type AvatarAccessoriesType = (typeof AVATAR_ACCESSORIES_TYPES)[number];

export const AVATAR_EYE_TYPES = ["Close", "Cry", "Default", "Dizzy", "EyeRoll", "Happy", "Hearts", "Side", "Squint", "Surprised", "Wink", "WinkWacky"] as const;
export type AvatarEyeType = (typeof AVATAR_EYE_TYPES)[number];

export const AVATAR_EYEBROW_TYPES = [
  "Angry",
  "AngryNatural",
  "Default",
  "DefaultNatural",
  "FlatNatural",
  "FrownNatural",
  "RaisedExcited",
  "RaisedExcitedNatural",
  "SadConcerned",
  "SadConcernedNatural",
  "UnibrowNatural",
  "UpDown",
  "UpDownNatural",
] as const;
export type AvatarEyebrowType = (typeof AVATAR_EYEBROW_TYPES)[number];

export const AVATAR_MOUTH_TYPES = ["Concerned", "Default", "Disbelief", "Eating", "Grimace", "Sad", "ScreamOpen", "Serious", "Smile", "Tongue", "Twinkle", "Vomit"] as const;
export type AvatarMouthType = (typeof AVATAR_MOUTH_TYPES)[number];
