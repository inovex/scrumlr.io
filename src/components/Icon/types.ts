export type IconNames =
  | 'phase1-create'
  | 'phase1-communicate'
  | 'phase1-combine'
  | 'phase2-vote'
  | 'phase3-discuss'
  | 'phase3-define'
  | 'circle-arrow-left'
  | 'circle-arrow-right'
  | 'circle-add'
  | 'circle'
  | 'more'
  | 'plus'
  | 'circle-selection'
  | 'circle-selection-grey'
  | 'logout'
  | 'download'
  | 'edit'
  | 'donate'
  | 'pencil'
  | 'focus'
  | 'close'
  | 'close20'
  | 'close-circle'
  | 'feedback'
  | 'key-up'
  | 'key-down'
  | 'key-esc'
  | 'settings'
  | 'board'
  | 'trash'
  | 'chevron-left'
  | 'chevron-right'
  | 'stack-top'
  | 'stack-mid'
  | 'stack-hover'
  | 'arrow-left'
  | 'arrow-right';
