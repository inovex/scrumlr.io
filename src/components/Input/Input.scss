@import '../../constants/style';

.input__wrapper {
  display: flex;
  flex-direction: column;

  flex: 1;

  padding-right: $default-margin;

  min-width: 256px;
}

.input__label {
  text-transform: uppercase;
  font-size: $theme-font-size;
  font-weight: 500;
  margin-bottom: $base-unit * 2;
}

.input__description {
  margin-top: $base-unit * 1.5;
  font-size: $theme-font-size - $base-unit / 2;
}

.input {
  flex: 1;
  background: none;

  box-sizing: border-box;

  border: none;
  border-radius: 0;
  outline: none;

  font-family: 'Heebo', sans-serif;
  font-size: 16px;
  letter-spacing: 1px;
  line-height: 24px;

  color: currentColor;
}

.input--underlined > .input {
  border-bottom: 1px solid $color-middle-gray;
}

.input--underlined > .input:focus {
  border-bottom: none;
}

.input--underlined > .input__underline {
  margin-top: -1px;
  margin-bottom: 0px;
}

.input--underlined > .input:focus + .input__underline {
  margin-top: -1px;
  margin-bottom: 0;
}

.input__underline {
  align-self: center;

  background-color: currentColor;
  height: 1px;
  transition: width 0.25s linear;
  width: 0;

  margin-top: 1px;
}

.input__focus-theme-mint > .input:hover::placeholder,
.input__focus-theme-mint > .input:focus::placeholder {
  color: $color-mint;
}

.input__focus-theme-mint > .input__underline {
  background-color: $color-mint;
}

.input:focus + .input__underline,
.input:hover + .input__underline {
  width: 100%;
}

.input:focus + .input__underline {
  margin-top: 0px;
  height: 2px;
}

.input::placeholder {
  font-weight: bold;
  color: $color-alternative-gray;
  text-transform: uppercase;

  transition: color .25s linear;
}

.input--inverted-placeholder::placeholder {
  color: $color-white;
  opacity: 0.8;
}