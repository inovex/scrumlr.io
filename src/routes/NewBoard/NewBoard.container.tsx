import { getVal, getFirebase } from 'react-redux-firebase';

import { Board, Boards, StoreState } from '../../types';
import { OwnNewBoardProps, StateNewBoardProps } from './NewBoard';
import { AuthProvider, instantiateAuthProviders } from '../../constants/Auth';
import { authController } from '../../controller/auth';
import { RetroMode } from '../../constants/mode';
import * as firebase from 'firebase/app';
import { Crypto } from '../../util/crypto';

function initialBoardConfig(
  user: firebase.User,
  mode: RetroMode,
  secure: boolean = false
): Promise<Board> {
  const crypto = new Crypto();

  return Promise.all([crypto.initKeypair(), crypto.initSymmetricKey()]).then(
    async () => {
      return {
        public: {
          config: {
            secure
          }
        },
        private: {
          config: {
            sorted: false,
            creatorUid: user.uid,
            guided: true,
            guidedPhase: 0,
            timerExpiration: null,
            created: new Date().toISOString(),
            showAuthor: false,
            mode
          },
          cards: {},
          users: {
            [user.uid]: {
              name: user.displayName || '',
              image: user.photoURL || '',
              ready: false,
              publicKey: crypto.getPublicKey()!!
            }
          },
          keyStore: {
            [crypto.getPublicKey()!!]: await crypto.exportSymmetricKey(
              crypto.getPublicKey()!!
            )
          },
          presence: {}
        }
      };
    }
  );
}

let onSignIn = false;

export function mapStateToProps(
  state: StoreState,
  ownProps: OwnNewBoardProps
): StateNewBoardProps {
  const { fbState } = state;
  const firebase = getFirebase();

  let uid: string | null = null;
  const auth = onSignIn ? {} : firebase.auth();

  const boards: Boards = getVal(fbState, 'data/boards', undefined);

  if (auth.currentUser) {
    uid = auth.currentUser.uid;
  }

  function onCreateNewBoard(mode: RetroMode, secure: boolean) {
    const auth = getFirebase().auth();
    initialBoardConfig(auth.currentUser!!, mode, secure).then(board => {
      getFirebase()
        .ref('/boards')
        .push(board)
        .then((item: any) => {
          const key = item.getKey();
          location.hash = `/board/${key}`;
        });
    });
  }

  function onLogin(email: string, mode: RetroMode, secure: boolean) {
    onSignIn = true;

    authController(firebase)
      .signInAnonymously(email)
      .then(() => {
        const { state = { referrer: null } } = ownProps.location;
        if (state.referrer) {
          location.assign(state.referrer);
        } else {
          onCreateNewBoard(mode, secure);
        }

        onSignIn = false;
      });
  }

  const onProviderLogin = (provider: AuthProvider) => () => {
    const AUTH_PROVIDERS = instantiateAuthProviders(firebase);
    firebase.auth().signInWithRedirect(AUTH_PROVIDERS[provider]);
  };

  function onLogout() {
    firebase
      .auth()
      .signOut()
      .then(() => {
        location.hash = '/new';
      });
  }

  return {
    boards,
    uid,
    onCreateNewBoard,
    onLogin,
    onProviderLogin,
    onLogout
  };
}
