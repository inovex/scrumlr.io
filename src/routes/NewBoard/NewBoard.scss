.new-board__input {
  border: none;
  border-bottom: 1px solid #adadad;
  font-size: 16px;
}

.new-board > * {
  margin: 10px;
}
