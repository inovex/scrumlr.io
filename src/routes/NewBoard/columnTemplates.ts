import {Color} from "constants/colors";

export const columnTemplates: {[id: string]: {name: string; description?: string; columns: {name: string; hidden: boolean; color: Color}[]}} = {
  leanCoffee: {
    name: "Lean Coffee",
    columns: [
      {name: "Lean Coffee", hidden: false, color: "backlog-blue"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  positiveNegative: {
    name: "Positive/Negative",
    columns: [
      {name: "Positive", hidden: false, color: "backlog-blue"},
      {name: "Negative", hidden: false, color: "lean-lilac"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  startStopContinue: {
    name: "Start/Stop/Continue",
    columns: [
      {name: "Start", hidden: false, color: "backlog-blue"},
      {name: "Stop", hidden: false, color: "grooming-green"},
      {name: "Continue", hidden: false, color: "online-orange"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  madSadGlad: {
    name: "Mad/Sad/Glad",
    columns: [
      {name: "Mad", hidden: false, color: "backlog-blue"},
      {name: "Sad", hidden: false, color: "online-orange"},
      {name: "Glad", hidden: false, color: "poker-purple"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  plusDelta: {
    name: "Plus/Delta",
    columns: [
      {name: "Plus", hidden: false, color: "backlog-blue"},
      {name: "Delta", hidden: false, color: "lean-lilac"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  kalm: {
    name: "KALM",
    description: "Keep, Add, Less, More",
    columns: [
      {name: "Keep", hidden: false, color: "backlog-blue"},
      {name: "Add", hidden: false, color: "retro-red"},
      {name: "Less", hidden: false, color: "grooming-green"},
      {name: "More", hidden: false, color: "poker-purple"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  fourL: {
    name: "4L",
    description: "Liked, Learned, Lacked, Longed for",
    columns: [
      {name: "Liked", hidden: false, color: "backlog-blue"},
      {name: "Learned", hidden: false, color: "retro-red"},
      {name: "Lacked", hidden: false, color: "lean-lilac"},
      {name: "Longed for", hidden: false, color: "poker-purple"},
      {name: "Actions", hidden: true, color: "planning-pink"},
    ],
  },
  swot: {
    name: "SWOT",
    description: "Strengths, Weaknesses, Opportunities, Threats",
    columns: [
      {name: "Strengths", hidden: false, color: "backlog-blue"},
      {name: "Weaknesses", hidden: false, color: "online-orange"},
      {name: "Opportunities", hidden: false, color: "grooming-green"},
      {name: "Threats", hidden: false, color: "retro-red"},
      {name: "Actions", hidden: true, color: "poker-purple"},
    ],
  },
};
