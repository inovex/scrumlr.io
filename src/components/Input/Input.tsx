import * as cx from 'classnames';
import * as React from 'react';

import './Input.scss';

export interface InputProps {
  invertPlaceholder?: boolean;
  showUnderline?: boolean;
  focusTheme?: string;
  id?: string;
  label?: string;
  description?: string;

  [key: string]: any;
}

export class Input extends React.Component<InputProps, {}> {
  static defaultProps: Partial<InputProps> = {
    invertPlaceholder: false,
    showUnderline: true
  };

  render() {
    const {
      label,
      description,
      className,
      showUnderline,
      invertPlaceholder,
      id,
      focusTheme,
      ...other
    } = this.props;

    return (
      <div
        className={cx('input__wrapper', className, {
          'input--underlined': showUnderline,
          [`input__focus-theme-${focusTheme}`]: Boolean(focusTheme)
        })}
      >
        {label && (
          <label htmlFor={id} className="input__label">
            {label}
          </label>
        )}
        <input
          {...other}
          id={id}
          className={cx('input', {
            'input--inverted-placeholder': invertPlaceholder
          })}
        />
        <span className={cx('input__underline')} />
        {description && (
          <span className="input__description">{description}</span>
        )}
      </div>
    );
  }
}

export default Input;
