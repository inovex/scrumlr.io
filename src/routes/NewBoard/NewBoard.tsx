import * as React from "react";
import Input from "@material-ui/core/Input";
import Button from "@material-ui/core/Button";
import {AuthenticationManager} from "utils/authentication/AuthenticationManager";
import {getRandomName} from "constants/Name";
import {RouteComponentProps} from "react-router";
import Parse from "parse";
import {API} from "api";
import {getColorForIndex} from "constants/colors";

export type NewBoardProps = RouteComponentProps;

function NewBoard(props: NewBoardProps) {
  const [name, setName] = React.useState(getRandomName());
  function handleChangeName(e: React.ChangeEvent<HTMLInputElement>) {
    setName(e.target.value);
  }

  async function onCreateBoard() {
    if (Parse.User.current()) {
      const boardId = await API.createBoard([
        {name: "Positive", hidden: false, color: getColorForIndex(0)},
        {name: "Negative", hidden: false, color: getColorForIndex(1)},
        {name: "Actions", hidden: true, color: getColorForIndex(2)},
      ]);
      props.history.push(`/board/${boardId}`);
    }
    // TODO report error
  }

  async function onLogin() {
    await AuthenticationManager.signInAnonymously(name);
    await onCreateBoard();
  }

  return (
    <div className="new-board">
      <Input
        className="new-board__input"
        defaultValue={name}
        type="text"
        onChange={handleChangeName}
        onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => {
          if (e.key === "Enter") {
            onLogin();
          }
        }}
        inputProps={{
          maxLength: 20,
        }}
      />
      <Button onClick={onLogin}>Login</Button>
    </div>
  );
}

export default NewBoard;
