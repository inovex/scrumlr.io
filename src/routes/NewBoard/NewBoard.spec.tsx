import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { mockStoreState } from '../../builder';
import { StoreState } from '../../types';
import { NewBoard, NewBoardProps } from './NewBoard';
import { mapStateToProps } from './NewBoard.container';
import Input from '../../components/Input';
const firebaseMock = require('react-redux-firebase');

// TODO: Write a mock function for this after Johanns Definitions have been merged.
const fbState = {
  auth: {
    uid: 'xNpM1E6XiigmfH7P8f42Vc3KyN02',
    displayName: 'somebody@example.com',
    photoURL:
      'https://www.gravatar.com/avatar/3e046b237bd8b23336231b3c0a577e56?s=32&d=retro',
    email: 'somebody@example.com',
    emailVerified: false,
    identifierNumber: null,
    isAnonymous: true,
    providerData: [],
    apiKey: 'someApiKey',
    appName: '[DEFAULT]',
    authDomain: 'somedomain.firebaseapp.com',
    redirectEventId: null
  },
  data: {
    boards: {
      '-KqhC0R4ywKluWRz0cZ8': {
        config: {
          created: '2017-08-04T12:24:32.064Z',
          creatorUid: 'xNpM1E6XiigmfH7P8f42Vc3KyN02',
          guided: true,
          guidedPhase: 0,
          sorted: false,
          users: {
            xNpM1E6XiigmfH7P8f42Vc3KyN02: {
              image:
                'https://www.gravatar.com/avatar/3e046b237cd8b23336231b3c0a577e56?s=32&d=retro',
              name: 'xx@xx.de',
              ready: false
            }
          }
        }
      },
      '-AqhC0R4ywKluWRz0cZ8': {
        config: {
          created: '2017-08-04T12:24:32.064Z',
          creatorUid: 'xNpM1E6XiigmfH7P8f42Vc3KyN02',
          guided: true,
          guidedPhase: 0,
          sorted: false,
          users: {
            xNpM1E6XiigmfH7P8f42Vc3KyN02: {
              image:
                'https://www.gravatar.com/avatar/3e046b237cd8b23336231b3c0a577e56?s=32&d=retro',
              name: 'xx@xx.de',
              ready: false
            }
          }
        }
      }
    }
  }
};

describe('<NewBoard />', () => {
  describe('mapStateToProps', () => {
    let state: StoreState;
    let ownProps: NewBoardProps;

    beforeEach(() => {
      state = mockStoreState({ fbState });
      firebaseMock.__setState(state.fbState);
      ownProps = {
        location: {
          state: {
            referrer: 'http://example.com/some/url'
          }
        }
      } as NewBoardProps;
    });

    it('should list all available boards', () => {
      const result = mapStateToProps(state, ownProps);
      expect(result.boards).toMatchSnapshot();
    });

    it('should get current user id', () => {
      const result = mapStateToProps(state, ownProps);
      expect(result.uid).toBe('xNpM1E6XiigmfH7P8f42Vc3KyN02');
    });

    it('should login user correctly', () => {
      const result = mapStateToProps(state, ownProps);
      const email = 'me@example.com';
      result.onLogin(email);
      const firebase = firebaseMock.getFirebase();
      expect(firebase.auth().signInAnonymously).toBeCalled();
      // TODO: can this be tested somehow?
      // expect(firebase.auth().currentUser.updateProfile).toBeCalled();
    });

    it('should sign out user correclty', () => {
      const result = mapStateToProps(state, ownProps);
      result.onLogout();
      const firebase = firebaseMock.getFirebase();
      expect(firebase.auth().signOut).toBeCalled();
    });
  });

  describe('dumb component', () => {
    let props: NewBoardProps;
    let shallowWrapper: ShallowWrapper<NewBoardProps, {}>;

    beforeEach(() => {
      props = {
        uid: null,
        boards: {},
        onLogin: jest.fn(),
        onProviderLogin: jest.fn(),
        onLogout: jest.fn(),
        onCreateNewBoard: jest.fn(),

        match: {
          params: {},
          isExact: true,
          path: '/new',
          url: '/new'
        },
        location: {
          state: {
            referrer: 'http://example.com/some/url'
          }
        } as any,
        history: null as any
      };
    });

    it('should not show the logout button if uid is falsy', () => {
      const uid = null;
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);
      const result = shallowWrapper.find('button.init-board__logout-btn');
      expect(result).toHaveLength(0);
    });

    it('should not show the logout button if uid is truthy', () => {
      const uid = 'someId';
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);
      const result = shallowWrapper.find('.new-board__logout-btn');
      expect(result).toHaveLength(1);
    });

    it('should trigger onLogout method if logout button is clicked', () => {
      const uid = 'someId';
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);
      shallowWrapper.find('.new-board__logout-btn').simulate('click');
      expect(props.onLogout).toBeCalled();
    });

    it('should show email field if user is not logged in', () => {
      const uid = null;
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);
      expect(shallowWrapper.find(Input)).toHaveLength(1);
    });

    it('should not show email field if user is logged in', () => {
      const uid = 'myUserId';
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);
      expect(shallowWrapper.find(Input)).toHaveLength(0);
    });

    it('should show login button if user is not logged in', () => {
      const uid = null;
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);

      const btn = shallowWrapper.find('.new-board__action-button');
      expect(btn.text()).toMatchSnapshot();
    });

    it('should trigger onLogin method if user is not logged in and press on login button', () => {
      const uid = null;
      const email = 'foo@bar.com';
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);

      const btn = shallowWrapper.find('.new-board__action-button');
      shallowWrapper.setState({ email });

      btn.simulate('click');
      expect(props.onLogin).toHaveBeenCalledWith(email);
    });

    it('should show create board button if user is logged in', () => {
      const uid = 'myUserId';
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);

      const btn = shallowWrapper.find('.new-board__action-button');
      expect(btn.text()).toMatchSnapshot();
    });

    it('should trigger onCreateNewBoard method if user is logged in and clicks on new board button', () => {
      const uid = 'myUserId';
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);

      const btn = shallowWrapper.find('.new-board__action-button');
      btn.simulate('click');
      expect(props.onCreateNewBoard).toHaveBeenCalled();
    });

    it('should update state if email address is entered', () => {
      const uid = null;
      shallowWrapper = shallow(<NewBoard {...props} uid={uid} />);

      //expect(shallowWrapper.instance().state).toEqual({ email: '' });

      shallowWrapper
        .find(Input)
        .simulate('change', { target: { value: 'abc' } });
      expect(shallowWrapper.instance().state).toEqual({ email: 'abc' });
    });
  });
});
