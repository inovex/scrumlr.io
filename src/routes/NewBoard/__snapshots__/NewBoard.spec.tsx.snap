// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewBoard /> dumb component should show create board button if user is logged in 1`] = `"Start"`;

exports[`<NewBoard /> dumb component should show login button if user is not logged in 1`] = `"Start"`;

exports[`<NewBoard /> mapStateToProps should list all available boards 1`] = `
Object {
  "-AqhC0R4ywKluWRz0cZ8": Object {
    "config": Object {
      "created": "2017-08-04T12:24:32.064Z",
      "creatorUid": "xNpM1E6XiigmfH7P8f42Vc3KyN02",
      "guided": true,
      "guidedPhase": 0,
      "sorted": false,
      "users": Object {
        "xNpM1E6XiigmfH7P8f42Vc3KyN02": Object {
          "image": "https://www.gravatar.com/avatar/3e046b237cd8b23336231b3c0a577e56?s=32&d=retro",
          "name": "xx@xx.de",
          "ready": false,
        },
      },
    },
  },
  "-KqhC0R4ywKluWRz0cZ8": Object {
    "config": Object {
      "created": "2017-08-04T12:24:32.064Z",
      "creatorUid": "xNpM1E6XiigmfH7P8f42Vc3KyN02",
      "guided": true,
      "guidedPhase": 0,
      "sorted": false,
      "users": Object {
        "xNpM1E6XiigmfH7P8f42Vc3KyN02": Object {
          "image": "https://www.gravatar.com/avatar/3e046b237cd8b23336231b3c0a577e56?s=32&d=retro",
          "name": "xx@xx.de",
          "ready": false,
        },
      },
    },
  },
}
`;
